lua_examples/luaOTA/_doTick.lua

```diff
@@ -0,0 +1,76 @@
+tmr.stop(0)--SAFETRIM 
+-- function _doTick(self)
+
+  -- Upvals
+  local self                = ...
+  local wifi,net            = wifi,net 
+  local sta                 = wifi.sta
+  local config,log,startApp = self.config,self.log,self.startApp
+  local tick_count          = 0
+  
+  local function socket_close(socket) --upval: self, startApp
+    if rawget(self,"socket") then
+      self.socket=nil                        -- remove circular reference in upval 
+      pcall(socket.close,socket)
+      return startApp("Unexpected socket close")
+    end
+  end
+
+  local function receiveFirstRec(socket, rec)  -- upval: self, crypto, startApp, tmr
+    local cmdlen   = (rec:find('\n',1, true) or 0) - 1
+    local cmd,hash = rec:sub(1,cmdlen-6), rec:sub(cmdlen-5,cmdlen)
+    if cmd:find('"r":"OK!"',1,true) or cmdlen < 16 or
+      hash ~= crypto.toHex(crypto.hash("MD5",self.secret .. cmd):sub(-3)) then
```

Comment:
md5 is a weak hash and this use is open to extension attacks (that is, given a signed command "foo", I can generate a signed command "foobar" without knowing the secret.)  Please use something stronger, like crypto.hmac('sha1', cmd, self.secret).  (Comment applies in several places.)

Consider using base64, rather than hex, for a slightly shorter hash, too?